tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -0,0 +1,83 @@
+// <copyright file="ProcessStartCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
+{
+    internal static class ProcessStartCommon
+    {
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.Process;
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProcessStartCommon));
+        internal const string OperationName = "command_execution";
+        internal const string ServiceName = "command";
+        internal const int MaxCommandLineLength = 4096;
+
+        internal static Scope CreateScope(ProcessStartInfo info)
+        {
+            if (info != null)
+            {
+                // In .NET framework, we need to have the UseShellExecute property set to false in order to use environment variables.
+#if !NETCOREAPP
+                return CreateScope(info.FileName, info.UseShellExecute ? null : info.Environment);
+#else
+                return CreateScope(info.FileName, info.Environment);
+#endif
+            }
+
+            return null;
+        }
+
+        internal static Scope CreateScope(string filename, IDictionary<string, string> environmentVariables)
+        {
+            var tracer = Tracer.Instance;
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this span
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                var variablesTruncated = EnvironmentVariablesScrubber.ScrubEnvironmentVariables(environmentVariables);
+                if (variablesTruncated?.Length > MaxCommandLineLength)
+                {
+                    variablesTruncated = Truncate(variablesTruncated, MaxCommandLineLength);
+                }
```

Comment:
`Truncate()` checks the string length, so this check is redundant.

```suggestion
                variablesTruncated = Truncate(variablesTruncated, MaxCommandLineLength);
```